standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreDefaultTransformer.java

```diff
@@ -695,7 +695,8 @@ public Database transformDatabase(Database db, List<String> processorCapabilitie
     if (!isTenantBasedStorage) {
       Path locationPath = Path.getPathWithoutSchemeAndAuthority(new Path(db.getLocationUri()));
       Path whRootPath = Path.getPathWithoutSchemeAndAuthority(hmsHandler.getWh().getWhRoot());
-      if (FileUtils.isSubdirectory(whRootPath.toString(), locationPath.toString())) { // legacy path
+      LOG.debug("Comparing DB and warehouse paths warehouse={} db.getLocationUri={}", whRootPath.toString(), locationPath.toString());
+      if (FileUtils.isSubdirectory(whRootPath.toString(), locationPath.toString()) || locationPath.equals(whRootPath)) { // legacy path
```

Comment:
It is unclear to me why we  set the managedLocationUri in this if block. It looks like if the db location is under warehouse location we will be setting managed location same as the db location. Why is that needed? A comment here about the reasoning for this would be very helpful.

Reply:
so prior to the introduction of managedlocation for db, while the exact semantics are unclear, the location was meant to be for managed tables. Although there is no code that enforces this strictly. So a legacy db object could have a location that could point anywhere. 
This fix it for backward compatibility. When a location is set to a managed location, we are ensuring that that location now becomes the managedlocation.

Reply:
Synced up with Naveen offline. I think it would be great to add a comment here explaining when the managed Location is set and when it is not. The patch itself looks good to me. Once we have a comment, I think we can merge it without the need for precommit (since there is only a comment which is added).